src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -859,6 +834,7 @@ def forward(self, inputs, targets):
 
         self.train_loader = train_loader or self.train_loader
         self.valid_loader = valid_loader or self.valid_loader
+        self._set_dataset_params()
```

Comment:
on each backward step? why?

Reply:
why not once in the beginning of train()

Reply:
?